crates/optimism/consensus/src/lib.rs

```diff
@@ -83,19 +97,67 @@ impl Consensus for OpBeaconConsensus {
         }
 
         // EIP-4895: Beacon chain push withdrawals as operations
-        if self.chain_spec.is_shanghai_active_at_timestamp(block.timestamp) {
-            validate_shanghai_withdrawals(block)?;
+        if self.chain_spec.is_shanghai_active_at_timestamp(block.timestamp) &&
+            block.body.withdrawals.as_ref().is_some_and(|withdrawals| !withdrawals.is_empty())
+        {
+            debug!(target: "op::consensus",
+                block_number=block.number,
+                err=%OpConsensusError::WithdrawalsNonEmpty,
+                "block failed validation",
+            );
+            return Err(ConsensusError::Other)
         }
 
         if self.chain_spec.is_cancun_active_at_timestamp(block.timestamp) {
             validate_cancun_gas(block)?;
         }
 
+        if self.chain_spec.is_isthmus_active_at_timestamp(block.timestamp) {
+            let storage_root_msg_passer = self
```

Comment:
this is problematic in this function,

because we use this when we validate a downloaded block, and accessing the latest state on disk might not exists.

need to figure out how to make this work.

could we in this pr introduce the isthmus check as a standalone function only and figure out where we actually need to check this separately?

Reply:
https://github.com/paradigmxyz/reth/pull/13311